### ISSUE TITLE
Query: subquery with orderby but without skip causes client eval - we could inject Skip(0) to allow for full translation

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#10102]
query:

```cs
                var query = from g in ctx.Gears
                            join w in ctx.Weapons.OrderBy(ww => ww.Name) on g.FullName equals w.OwnerFullName
                            select new { g, w.Name };
```

causes client join, issuing 2 queries to the server. This is because sql doesn't support orderby in a subquery without skip/take

However:

```cs
                var query = from g in ctx.Gears
                            join w in ctx.Weapons.OrderBy(ww => ww.Name).Skip(0) on g.FullName equals w.OwnerFullName
                            select new { g, w.Name };
```

translates fully into:

```sql
SELECT [g].[Nickname], [g].[SquadId], [g].[AssignedCityName], [g].[CityOrBirthName], [g].[Discriminator], [g].[FullName], [g].[HasSoulPatch], [g].[LeaderNickname], [g].[LeaderSquadId], [g].[Rank], [t].[Name]
FROM [Gears] AS [g]
INNER JOIN (
    SELECT [ww].*
    FROM [Weapons] AS [ww]
    ORDER BY [ww].[Name]
    OFFSET @__p_0 ROWS
) AS [t] ON [g].[FullName] = [t].[OwnerFullName]
WHERE [g].[Discriminator] IN (N'Officer', N'Gear')
```

We should consider recognizing this pattern and injecting Skip(0) to make this query better.

**Comment by Muppets** [comment]
@maumar @ajcvickers I've opened a WIP PR for this and could do with some feedback on the changes and some help on the last few unit tests I can't quite get my head around.

**Comment by maumar** [comment]
fixed in 4a2b27f8fbf484fa2ea09bfb7fb785a170189b42
